CACHING.md

```diff
@@ -0,0 +1,351 @@
+# Cache Recipes
+
+This document lists some of the strategies (and example workflows if possible) which can be used
+
+- to solve some common use cases
+- to effectively leverage the step inputs and outputs
+
+## Strategically using keys
+
+```yaml
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    - uses: actions/cache@v3
+      with:
+        key: ${{ some-metadata }}-cache
+```
+
+In your workflows, you can use different strategies to name your key depending on your use case so that the cache is scoped properly based on the need. If you wish to create OS specific caches, or caches based on the lockfiles, commit SHA, workflow run id, etc. then you can generate the keys dynamically at run-time. Below are some of the tips to strategically name your cache using the [cache](https://github.com/actions/cache) or [restore](https://github.com/actions/cache/tree/main/restore) action.
```

Comment:
Shortening the sentence to try to keep it simple. Removed reference to actions as that is quite obvious. No?
```suggestion
In your workflows, you can use different strategies to name your key depending on your use case so that the cache is scoped appropriately for your need. For example, you can have cache specific to OS, or based on the lockfile or commit SHA or even workflow run.
```